### PR TITLE
Ensure RefSeq parser creates non-empty CDS regions

### PR DESCRIPTION
**Pull request by roland-ewald**
This seems to be a new regression in `v0.28`, introduced via #435

## How to reproduce

1. run `java jannovar-cli-0.28.jar download -d hg38/refseq_curated` 
2. annotate a VCF: all variants will be classified as 'non-coding' and annotated accordingly (with HGVS cDNA such as `n.1234` etc.)
3. load `JannovarData` stored in `./data/hg38_refseq_curated.ser` and inspect it (e.g. with a debugger): the size of the CDS regions in the `TranscriptModel`s is 0; they are all empty and defined as initialized per default in `RefSeqParser#mapParentIdsToTranscriptModelsWithTxRegion`.

##  Problem

The problem is that the parser currently skips over all `CDS` records, which (as correctly pointed out in the comments) have no transcript IDs: https://github.com/charite/jannovar/blob/368762d1730d543ac304a8d9f1fc0f1aaf952ed4/jannovar-core/src/main/java/de/charite/compbio/jannovar/impl/parse/refseq/RefSeqParser.java#L230

## Fix

Besides fixing this bug, this PR also changes the behavior of `RefSeqParser` so that it generates an `IllegalStateException` if an empty CDS region is encountered for coding transcripts (`NM_*` / `XM_*`). This should help to avoid a silent failure due to another refactoring (or incomplete data) in the future.

The test data (the `gff.gz` file) had to be adapted (the CDS region data was missing) and the test is extended to check the generated regions for one example (`NM_001636.3`).

### CODE REVIEW

**Comment by visze** [comment]
Looks good to me. can you do a pull request on develop? think we will update jannovar to a newer version rather than creating a new fix in master

**Comment by roland-ewald** [comment]
@visze Sure! Sorry. Changed the base to `develop` now.

File: jannovar-core/src/main/java/de/charite/compbio/jannovar/impl/parse/refseq/RefSeqParser.java [hunk 1]
```diff
@@ -227,7 +221,7 @@ private void loadFASTA(Map<String, TranscriptModelBuilder> builders, String path
 			while ((record = parser.next()) != null) {
 				numRecords++;
 				String transcriptId = record.getAttributes().get("transcript_id");
-				if (onlyCurated && (transcriptId == null || transcriptId.startsWith("X"))) {
+				if (onlyCurated && (transcriptId != null && transcriptId.startsWith("X"))) {
```
Comment (visze) [inline jannovar-core/src/main/java/de/charite/compbio/jannovar/impl/parse/refseq/RefSeqParser.java]: can you remove the non-necessary brackets?

Comment (roland-ewald) [reply jannovar-core/src/main/java/de/charite/compbio/jannovar/impl/parse/refseq/RefSeqParser.java]: done

**Comment by roland-ewald** [comment]
@visze added entry to `changelog`

**Pull request merged**